### PR TITLE
Update Scale version check in Snapshot/cloning to check v5.1.2.1

### PR DESCRIPTION
**Pull request by madhuthorat**
as minimum required Scale version.

Signed-off-by: Madhu Thorat <madhu.punjabi@in.ibm.com>

### CODE REVIEW

File: driver/csiplugin/controllerserver.go [hunk 1]
```diff
@@ -832,14 +832,14 @@ func (cs *ScaleControllerServer) checkMinScaleVersion(conn connectors.SpectrumSc
...
-	versionCheck, err := cs.checkMinScaleVersion(conn, "5110")
+	/* Verify Spectrum Scale Version is not below 5.1.2-1 */
+	versionCheck, err := cs.checkMinScaleVersion(conn, "5121")
 	if err != nil {
 		return err
 	}
 
 	if !versionCheck {
-		return status.Error(codes.FailedPrecondition, "the minimum required Spectrum Scale version for snapshot support with CSI is 5.1.1-0")
+		return status.Error(codes.FailedPrecondition, "the minimum required Spectrum Scale version for snapshot support with CSI is 5.1.2-1")
```
Comment (Jainbrt) [inline driver/csiplugin/controllerserver.go:842]: We have to enhance this error message to "the minimum required Spectrum Scale version for CSI snapshot & cloning support is 5.1.1-0"

**Review by Jainbrt — CHANGES_REQUESTED** [review]

File: driver/csiplugin/controllerserver.go [hunk 2]
```diff
@@ -921,9 +934,9 @@ func (cs *ScaleControllerServer) validateSrcVolumeID(vID *scaleVolId, scVol *sca
 	}
 
 	// This is kind of snapshot restore
```
Comment (Jainbrt) [inline driver/csiplugin/controllerserver.go:936]: what does this comment mean here?

Comment (madhuthorat) [reply driver/csiplugin/controllerserver.go:936]: In Snapshot restore, source is snapshot and destination is PVC. In Volume cloning, source is PVC and destination is PVC. So as copy to PVC happens, it is mentioned as kind of snapshot restore.

**Review by Jainbrt — APPROVED** [review]
LGTM

**Pull request merged**